Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1

```diff
@@ -0,0 +1,642 @@
+        if($quiet -eq $true){
+            $selectedSite = $sites | Where-Object { $_.Name -eq $selectedSiteName }
+        }else{
+            if($sites -is [array]){
+                $sites |
+                    ForEach-Object {New-Object PSCustomObject -Property @{
+                        'Id'=$_.id;
+                        'Name'=$_.name;
+                        'Physical Path'=[System.Environment]::ExpandEnvironmentVariables($_.physicalPath);
+                        'Bindings'=$_.bindings;
+                    };} |
+                    Format-Table Id,Name,'Physical Path',Bindings -AutoSize | Out-Host
+
+                do {
+                    $selectedSiteId = Read-Host "Select a web site by Id"
+                    $selectedSite = $sites[$selectedSiteId - 1]
+                    if([string]::IsNullOrEmpty($selectedSiteId)){
+                        Write-DosMessage -Level "Information" -Message "You must select a web site."
+                    }
+                } while ([string]::IsNullOrEmpty($selectedSiteId))
+                
+            }else{
+                $selectedSite = $sites
+            }
+        }
+        if($null -eq $selectedSite){
+            throw "Could not find selected site."
+        }
+        if($selectedSite.Name){ Add-InstallationSetting "identity" "siteName" $selectedSite.Name | Out-Null }
+
+        return $selectedSite
+
+    }catch{
+        Write-DosMessage -Level "Error" -Message "Could not select a website."
+        throw
+    }
+}
+
+function New-SigningAndEncryptionCertificate([string] $subject, [string] $certStorelocation)
+{
+    $cert = New-SelfSignedCertificate -Type Custom -Subject $subject -KeyUsage DataEncipherment, DigitalSignature -KeyAlgorithm RSA -KeyLength 2048 -CertStoreLocation $certStoreLocation
+    return $cert
+}
+
+function Get-Certificates([string] $primarySigningCertificateThumbprint, [string] $encryptionCertificateThumbprint, [bool] $quiet){
+    if(Test-ShouldShowCertMenu -primarySigningCertificateThumbprint $primarySigningCertificateThumbprint `
+                                -encryptionCertificateThumbprint $encryptionCertificateThumbprint `
+                                -quiet $quiet){
+        try{
+            $today = Get-Date
+            $allCerts = Get-CertsFromLocation Cert:\LocalMachine\My
+            $index = 1
+            $attempts = 1
+            $allCerts | 
+                Where-Object { $_.NotAfter -ge $today -and $_.NotBefore -le $today } |
+                ForEach-Object {New-Object PSCustomObject -Property @{
+                'Index'=$index;
+                'Subject'= $_.Subject; 
+                'Name' = $_.FriendlyName; 
+                'Thumbprint' = $_.Thumbprint; 
+                'Expiration' = $_.NotAfter
+                };
+                $index ++} |
+                Format-Table Index,Name,Subject,Expiration,Thumbprint  -AutoSize | Out-Host
+            do {
+                if($attempts -gt 10){
```

Comment:
Why does this function have an attempt counter?

Reply:
So that I can test it in pester without getting into an infinite loop :-).

Reply:
I haven't figured out how to create a mock that does something different the second time the mock is called with the same parameters...